libopenage/CMakeLists.txt

```diff
@@ -163,3 +181,33 @@ if(NOT APPLE)
 endif()
 
 install(TARGETS libopenage DESTINATION lib LIBRARY DESTINATION lib)
+
+
+# add custom commands to generate pxd files from registered header files
+get_property(PXDGEN_SOURCES GLOBAL PROPERTY SFT_PXDGEN_SOURCES)
+foreach(PXDGEN_SOURCE ${PXDGEN_SOURCES})
+	file(RELATIVE_PATH PXDGEN_SOURCE_RELATIVE_PATH "${CMAKE_SOURCE_DIR}" "${PXDGEN_SOURCE}")
+	string(REGEX REPLACE "\\.h$" ".pxd" GENERATED_PXD_FILE "${PXDGEN_SOURCE_RELATIVE_PATH}")
+	set(GENERATED_PXD_FILE "${CMAKE_BINARY_DIR}/${GENERATED_PXD_FILE}")
+
+	add_custom_command(OUTPUT "${GENERATED_PXD_FILE}"
+		COMMAND "${PYTHON}" -m buildsystem.pxdgen "${PXDGEN_SOURCE}" "${GENERATED_PXD_FILE}"
+		DEPENDS "${PXDGEN_SOURCE}"
+		WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
+		COMMENT "Generating pxd file for ${PXDGEN_SOURCE_RELATIVE_PATH}"
+	)
+	list(APPEND GENERATED_PXD_FILES "${GENERATED_PXD_FILE}")
+endforeach()
+set_source_files_properties(${GENERATED_PXD_FILES} PROPERTIES GENERATED ON)
+
+# This target is a dependency for the build_ext and therefore ensuring that all pxd files are generated
+# before the python extensions are built.
+add_custom_target(pxdgen
+	DEPENDS ${GENERATED_PXD_FILES}
+)
+
+add_custom_target(cleanpxdgen
+	COMMAND find "${CMAKE_BINARY_DIR}" -name "*.pxd" -type f -print -delete
+	COMMAND find "${CMAKE_BINARY_DIR}" -name "__init__.py" -type f -print -delete
+	WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
+)
```

Comment:
the pxdgen stuff once again has to be in `buildsystem/`, not in the library folder